packages/js/product-editor/src/components/variations-table/variation-actions-menu/variation-actions-menu.tsx

```diff
@@ -67,14 +25,16 @@ export function VariationActionsMenu( {
 	function handlePrompt(
 		propertyName: keyof ProductVariation,
 		label: string = __( 'Enter a value', 'woocommerce' ),
-		parser: ( value: string ) => unknown = ( value ) => value
+		parser: ( value: string ) => unknown = ( value ) => value,
+		additionalProps: Partial< ProductVariation > = {}
```

Comment:
I think that if the `parser` function returns the whole variation we don't need this `additionalProps` anymore right. Also the `handlePrompt` function does not need to know about the `propertyName` in that case. I was having a similar thinking doing Shipping https://github.com/woocommerce/woocommerce/pull/39925/files#diff-2f82b22147fa0e6b6dd8a3d351ea8856cbdfbd185ee4d79ca54c147373d0c0ccR64 and I thought changing it gave us more flexibility, for example, your case here.

Before
```
onChange( {
	[ propertyName ]: parser( value.trim() ),
	...additionalProps,
} );
```

After

```
onChange( parser( value.trim() ) );
```

Reply:
@mdperez86 updated as part of https://github.com/woocommerce/woocommerce/pull/39935/commits/772842f4f51fb55dc4268cabfecbcb5b03194e7e